test/redis_test.rb

```diff
@@ -51,6 +51,29 @@ def test_connection_errors_open_the_circuit
     end
   end
 
+  def test_dns_resolution_errors_in_ruby_driver_open_the_circuit
+    redis_options = { host: "invalid-dns-name", semian: SEMIAN_OPTIONS, reconnect_attempts: 0 }
+
+    clients = [
+      Redis.new(driver: 'ruby', **redis_options),
+      # Redis.new(driver: 'hiredis', **redis_options)
```

Comment:
well the test is gonna fail :) and I don't like that ugly hack where we look at the message :/